### ISSUE TITLE
Elasticsearch exits after installation

### ISSUE THREAD

**Issue by NudeWeb** [bitnami/bitnami-docker-magento#125]
**Description**

Cannot complete magento installation because Elasticsearch exits after 1 sec. 

**Steps to reproduce the issue:**

$ curl -sSL https://raw.githubusercontent.com/bitnami/bitnami-docker-magento/master/docker-compose.yml > docker-compose.yml
$ docker-compose up -d

**Describe the results you received:**

Docker ps 
CONTAINER ID        IMAGE                     COMMAND                  CREATED             STATUS                      PORTS                                      NAMES
e9f8cdca082d        bitnami/magento:2         "/app-entrypoint.sh …"   18 seconds ago      Up 16 seconds               0.0.0.0:80->80/tcp, 0.0.0.0:443->443/tcp   containers_magento_1
0b4a50e5d97a        bitnami/elasticsearch:6   "/entrypoint.sh /run…"   20 seconds ago      Exited (1) 17 seconds ago                                              containers_elasticsearch_1
8dfd0e167ce0        bitnami/mariadb:10.2      "/entrypoint.sh /run…"   20 seconds ago      Up 18 seconds               3306/tcp                                   containers_mariadb_1

**Describe the results you expected:**

All 3 containers up

**Additional information you deem important (e.g. issue happens only occasionally):**

**Version**

- Output of `docker version`:

```
Client: Docker Engine - Community
 Version:           19.03.5
 API version:       1.40
 Go version:        go1.12.12
 Git commit:        633a0ea838
 Built:             Wed Nov 13 07:29:52 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.5
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.12
  Git commit:       633a0ea838
  Built:            Wed Nov 13 07:28:22 2019
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.2.10
  GitCommit:        b34a5c8af56e510852c35414db4c1f4fa6172339
 runc:
  Version:          1.0.0-rc8+dev
  GitCommit:        3e425f80a8c931f88e6d94a8c831b9d5aa481657
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

- Output of `docker info`:

```
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: b34a5c8af56e510852c35414db4c1f4fa6172339
 runc version: 3e425f80a8c931f88e6d94a8c831b9d5aa481657
 init version: fec3683
 Security Options:
  apparmor
  seccomp
   Profile: default
 Kernel Version: 4.15.0-50-generic
 Operating System: Ubuntu 18.04.2 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 8
 Total Memory: 47.16GiB
 Name: ruud
 ID: X4EA:LQCU:KZEJ:FWQH:QMFQ:SCJB:O3XS:4XHM:IIMG:DEZ6:EWW4:3JSK
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

- Output of `docker-compose version` (if applicable):

```
docker-compose version 1.25.1, build a82fef07
docker-py version: 4.1.0
CPython version: 3.7.4
OpenSSL version: OpenSSL 1.1.0l  10 Sep 2019
```

**Additional environment details (AWS, VirtualBox, Docker for MAC, physical, etc.):**
Running on a virtual linux machine, Ubuntu

**Comment by alemorcuq** [comment]
Hi, @NudeWeb.

Can you share the logs of the Elasticsearch container? From my experience, Elasticsearch may be exitting because you have to change a kernel setting in your host (`fs.file-max` to at least 65536), and if that's the case the logs should reflect that.

**Comment by NudeWeb** [comment]
Thanks @alemorcuq that solved the elasticsearch problem. However the setup still failed to complete. When I browse to the site via the IP address it redirects to http://127.0.0.1/index.php/ 
Any idea why?

**Comment by alemorcuq** [comment]
Could you share the logs, @NudeWeb?

**Comment by octohedron** [comment]
Steps to fix this issue on `Ubuntu 19.10`

```
sudo vi /etc/security/limits.conf
```

Add this at the end

```
root soft  nofile 1000000 # or your desired amount
root hard  nofile 1000000 # or your desired amount
```

Then add the following line to: `/etc/pam.d/common-session`

```
session required pam_limits.so
```

Might need to also add this to your `/etc/sysctl.conf`
```
fs.file-max = 1000000
vm.max_map_count=262144
```
Reboot and see if it works

```
docker run bitnami/elasticsearch:6
```

**Comment by rafariossaa** [comment]
Hi @octohedron 
  Thanks for sharing your solution.
